editor/test/integration/editor_extensions_test.clj

```diff
@@ -375,20 +377,158 @@
+        (is (thrown? LuaError (coerce coerce/function inc)))))
+
+    (testing "wrap-with-pred"
+      (let [with-pred (coerce/wrap-with-pred coerce/integer even? "should be even")]
+        (is (= 2 (coerce with-pred 2)))
+        (is (thrown? LuaError (coerce with-pred 1)))
+        (is (thrown? LuaError (coerce with-pred "not an int")))))
+
+    (testing "coll-of"
+      (is (= [] (coerce (coerce/coll-of coerce/string) {:a 1}))) ;; non-integer keys are ignored
+      (is (= [] (coerce (coerce/coll-of coerce/string) {})))
+      (is (= ["a"] (coerce (coerce/coll-of coerce/string) {1 "a" :key "val"})))
+      (is (= ["a"] (coerce (coerce/coll-of coerce/string) {1 "a" -1 "b" 0 "c"}))) ;; non-positive keys are ignored
+      (is (= [1 2 3] (coerce (coerce/coll-of coerce/integer) [1 2 3])))
+      (is (= [1 2 3 nil 5] (coerce (coerce/coll-of coerce/any) [1 2 3 nil 5])))
+      (is (= [1 2 3 nil 5] (coerce (coerce/coll-of (coerce/one-of coerce/integer coerce/null)) [1 2 3 nil 5])))
+      (is (thrown? LuaError (coerce (coerce/coll-of coerce/integer) [1 2 3 nil 5])))
+      (is (= [nil nil nil nil 5] (coerce (coerce/coll-of (coerce/one-of coerce/integer coerce/null)) {5 5})))
+      (is (thrown? LuaError (coerce (coerce/coll-of coerce/integer) {5 5})))
+      (testing "min-count"
+        (is (= [1 2] (coerce (coerce/coll-of coerce/any :min-count 2) [1 2])))
+        (is (= [1 2 3] (coerce (coerce/coll-of coerce/any :min-count 2) [1 2 3])))
+        (is (thrown? LuaError (coerce (coerce/coll-of coerce/any :min-count 2) [1]))))
+      (testing "distinct"
+        (is (= [1 2 3] (coerce (coerce/coll-of coerce/any :distinct true) [1 2 3])))
+        (is (thrown? LuaError (coerce (coerce/coll-of coerce/any :distinct true) [1 2 1])))))
+
+    (testing "record"
+      (is (= {:a 1} (coerce (coerce/record :req {:a coerce/integer}) {:a 1})))
+      (is (= {"a" 1} (coerce (coerce/record :req {"a" coerce/integer}) {:a 1})))
+      (is (thrown? LuaError (coerce (coerce/record :req {:a coerce/integer}) {})))
+      (is (thrown? LuaError (coerce (coerce/record :req {:a coerce/integer}) {:a "not-an-int"})))
+      (is (thrown? LuaError (coerce (coerce/record :req {:a coerce/integer}) "not a table")))
+      (is (= {} (coerce (coerce/record :opt {:a coerce/integer}) {})))
+      (is (= {} (coerce (coerce/record :opt {:a coerce/integer}) {:x 1 :y 2})))
+      (is (thrown? LuaError (coerce (coerce/record :opt {:a coerce/integer}) {:x 1 :y 2 :a "not-an-int"})))
+      (is (= {:a 1} (coerce (coerce/record :opt {:a coerce/integer}) {:x 1 :y 2 :a 1})))
+      (is (= {:a 1} (coerce (coerce/record
+                              :req {:a coerce/integer}
+                              :opt {:b coerce/integer})
+                            {:a 1})))
+      (is (= {:a 1 :b 2} (coerce (coerce/record
+                                   :req {:a coerce/integer}
+                                   :opt {:b coerce/integer})
+                                 {:a 1 :b 2}))))
```

Comment:
Similarly, `record` seems like we might want to use for coercing into Clojure record types at some point? Maybe `coece/hashmap` would be better?

Reply:
I wanted to avoid the word `map` because we might want to introduce `(map-of key-coercer val-coercer)` later for homogeneous maps. I agree that `record` is not a very good term either. Clojure spec uses the term `keys`. Malli uses calls these `map`, so maybe it's fine? Yeah, I'll rename it to `hashmap` :)